### PR TITLE
Add components for AI2 project icons.

### PR DESCRIPTION
**Pull request by codeviking**
This adds components that render individual AI2 project icons
as SVG.

I added the `ai2/projects` directory in `components/`, given the
component collection is getting large. I also added `TODO`s to
move `AI2Logo` and `AI2Banner` into `ai2/` for consistency -- that
said doing so now would make this non-backwards compatible and only
necessitate a bunch of work for our users.

### CODE REVIEW

File: components/button.tsx [hunk 1]
```diff
@@ -136,22 +105,14 @@ export const Button = styled(AntButton).attrs<{ type?: string }>((props: Compone
...
             svg,
             path {
                 color: ${props =>
                     toHexIfDefined(
-                        valueOrDefault(
-                            props,
-                            b => b.disabled.color,
-                            b => b.disabled.contrastColor
-                        )
+                        valueOrDefault(props, b => b.disabled.color, b => b.disabled.contrastColor)
```
Comment (codeviking) [inline components/button.tsx]: The changes in this file all resulted from `yarn lint:fix`.

File: components/ai2/projects/icon.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+
+interface IconProps {
+    className?: string;
+    width?: string;
+    height?: string;
+}
+
+const defaults = {
+    SIZE: 32,
```
Comment (jonborchardt) [inline components/ai2/projects/icon.tsx]: get from defaulttheme?

File: components/ai2/projects/icon.tsx [hunk 3]
```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+
+interface IconProps {
```
Comment (jonborchardt) [inline components/ai2/projects/icon.tsx]: this interface belongs in a more general location

File: components/ai2/projects/icon.tsx [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    className?: string;
+    width?: string;
+    height?: string;
+}
+
+const defaults = {
```
Comment (jonborchardt) [inline components/ai2/projects/icon.tsx]: as do these defaults

File: components/ai2/projects/icon.tsx (hunk 3 above)
Comment (codeviking) [reply components/ai2/projects/icon.tsx]: I don't disagree, but I'll let that generalization follow later when / if.

File: components/ai2/projects/icon.tsx (hunk 2 above)
Comment (codeviking) [reply components/ai2/projects/icon.tsx]: In this case I'm not sure.

The icons are, natively, rendered at 32 x 32. So this feels like a more specific thing. I'm not sure though. We can always re-evaluate down the road.

File: components/ai2/projects/icon.tsx (hunk 4 above)
Comment (codeviking) [reply components/ai2/projects/icon.tsx]: ☝️

**Pull request merged**